torch/fx/passes/split_module.py

```diff
@@ -94,6 +94,7 @@ def record_cross_partition_use(def_node : torch.fx.node.Node, use_node : Optiona
         partition = partitions[partition_name]
         for input in partition.inputs:
             placeholder = partition.graph.placeholder(input)
+            placeholder.meta = orig_nodes[input].meta.copy()
```

Comment:
Maybe we want to do `copy.deepcopy(orig_nodes[input].meta)` in case someone mutates the meta after graph splitting

(maybe in 10 years i'll write a PEP for copy-on-write ownership in python :p )

Reply:
@jamesr66a I noticed we aren't using deep copy in graph's `node_copy()`: 
https://github.com/pytorch/pytorch/blob/bd3c63aeeb72b565fec707ce8af71230a0727e21/torch/fx/graph.py#L683

WDYT, should I update it in this PR as well, or just leave as is?

Reply:
I'm hitting issues due to shape prop's TensorMetadata's `torch.memory_format` not working correctly with `deepcopy`. I'm going to and as is since I think moving to deepcopy requires more code changes outside the scope of this PR, and as mentioned there are other places like in `node_copy()` we will want to move to `deepcopy` as well.